put/baton.go

```diff
@@ -85,6 +86,34 @@ func GetBatonHandler() (*Baton, error) {
 	return &Baton{}, err
 }
 
+// GetBatonHandlerWithMetaClient returns a Handler that uses Baton to interact
+// with iRODS and contains a meta client for interacting with metadata. If you
+// don't have baton-do in your PATH, you'll get an error.
```

Comment:
There should be some docs and, probably for GetBatonHandler() as well, and probably for server.Config.StorageHandler that makes it explicit you need to call this and not GetBatonHandler() for remove functionality to work.

Is there a better way of arranging things so there aren't 2 GetBaton...() methods? Or some other way of avoiding possible mistakes like this?